openupgrade_scripts/scripts/payment/17.0.2.0/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,338 @@
+NEW ir.rule: payment.payment_capture_wizard_rule (noupdate)
+# NOTHING TO DO
+
+DEL ir.rule: payment.payment_transaction_user_rule (noupdate)
+# DONE: safely delete in post-migration
+
+NEW ir.ui.view: payment.company_mismatch_warning
+NEW ir.ui.view: payment.form
+NEW ir.ui.view: payment.form_icon
+NEW ir.ui.view: payment.form_logo
+NEW ir.ui.view: payment.method_form
+NEW ir.ui.view: payment.payment_capture_wizard_view_form
+NEW ir.ui.view: payment.payment_method_form
+NEW ir.ui.view: payment.payment_method_kanban
+NEW ir.ui.view: payment.payment_method_search
+NEW ir.ui.view: payment.payment_method_tree
+NEW ir.ui.view: payment.portal_my_home_payment
+NEW ir.ui.view: payment.submit_button
+NEW ir.ui.view: payment.summary_item
+NEW ir.ui.view: payment.token_form
+# NOTHING TO DO
+
+DEL ir.ui.view: payment.checkout
+DEL ir.ui.view: payment.footer
+DEL ir.ui.view: payment.icon_list
+DEL ir.ui.view: payment.manage
+DEL ir.ui.view: payment.payment_icon_form
+DEL ir.ui.view: payment.payment_icon_tree
+DEL ir.ui.view: payment.verified_token_checkmark
+NEW onboarding.onboarding.step: payment.onboarding_onboarding_step_payment_provider (noupdate)
+DEL payment.icon: payment.payment_icon_airtel_money
+DEL payment.icon: payment.payment_icon_apple_pay
+DEL payment.icon: payment.payment_icon_barter_by_flutterwave
+DEL payment.icon: payment.payment_icon_bbva_bancomer
+DEL payment.icon: payment.payment_icon_cc_american_express
+DEL payment.icon: payment.payment_icon_cc_bancontact
+DEL payment.icon: payment.payment_icon_cc_cirrus
+DEL payment.icon: payment.payment_icon_cc_codensa_easy_credit
+DEL payment.icon: payment.payment_icon_cc_diners_club_intl
+DEL payment.icon: payment.payment_icon_cc_discover
+DEL payment.icon: payment.payment_icon_cc_eps
+DEL payment.icon: payment.payment_icon_cc_giropay
+DEL payment.icon: payment.payment_icon_cc_ideal
+DEL payment.icon: payment.payment_icon_cc_jcb
+DEL payment.icon: payment.payment_icon_cc_maestro
+DEL payment.icon: payment.payment_icon_cc_mastercard
+DEL payment.icon: payment.payment_icon_cc_p24
+DEL payment.icon: payment.payment_icon_cc_rupay
+DEL payment.icon: payment.payment_icon_cc_unionpay
+DEL payment.icon: payment.payment_icon_cc_visa
+DEL payment.icon: payment.payment_icon_cc_webmoney
+DEL payment.icon: payment.payment_icon_cc_western_union
+DEL payment.icon: payment.payment_icon_citibanamex
+DEL payment.icon: payment.payment_icon_kbc
+DEL payment.icon: payment.payment_icon_mada
+DEL payment.icon: payment.payment_icon_mpesa
+DEL payment.icon: payment.payment_icon_mtn_mobile_money
+DEL payment.icon: payment.payment_icon_paypal
+DEL payment.icon: payment.payment_icon_sadad
+DEL payment.icon: payment.payment_icon_sepa
+# NOTHING TO DO
+
+NEW payment.method: payment.payment_method_7eleven (noupdate)
+NEW payment.method: payment.payment_method_ach_direct_debit (noupdate)
+NEW payment.method: payment.payment_method_acss_debit (noupdate)
```

Comment:
This record has been removed https://github.com/odoo/odoo/commit/4b067a43593f3e2153be99d60cef69302601b6ee

Reply:
Ok nevermind won't affect the migration process

Reply:
Hello, it seems that if it affects, it gives error in https://github.com/OCA/OpenUpgrade/blob/17.0/openupgrade_scripts/scripts/payment/17.0.2.0/noupdate_changes.xml#L43

```
... OpenUpgrade: payment: error in migration script /opt/odoo/auto/addons/openupgrade_scripts/scripts/payment/17.0.2.0/post-migration.py: while parsing /opt/odoo/auto/addons/openupgrade_scripts/scripts/payment/17.0.2.0/noupdate_changes.xml:42, somewhere inside
....
External ID not found in the system: payment.payment_method_acss_debit
...
ValueError: <class 'ValueError'>: "External ID not found in the system: payment.payment_method_acss_debit" while evaluating
```

Reply:
Hi @duong77476-viindoo thank you for initializing these migration scripts.
I reproduce the error above, but you are right that it is not related to upgrade_analysis files but the noupdate_changes.xml need to be updated here : https://github.com/OCA/OpenUpgrade/blob/b2f4cfe3f6f43676d9e4dfdf28ea56bd272a6a3c/openupgrade_scripts/scripts/payment/17.0.2.0/noupdate_changes.xml#L43
to remove ref('payment.payment_method_acss_debit'), which was removed in commit listed by @ecino above, after you created the migration script (commit is from end of July 24)

Reply:
Yes, thank you. But currently me and my company stop doing migration, we don't have enough resource and will come back when we have more employees i guess. You can create another pr from this and ping me to close